crates/usvg/src/writer.rs

```diff
@@ -1228,3 +1369,274 @@ fn write_num(num: f32, buf: &mut Vec<u8>, precision: u8) {
 
     write!(buf, "{}", v).unwrap();
 }
+
+/// Write all of the tspan attributes except for baseline_shift and decorations.
+fn write_span(
+    is_clip_path: bool,
+    writer_context: &mut WriterContext,
+    xml: &mut XmlWriter,
+    text: &Text,
+    char_offset: &mut usize,
+    chunk: &TextChunk,
+    span: &TextSpan,
+) {
+    xml.start_svg_element(EId::Tspan);
+
+    write_font_family(xml, span);
+    write_font_style(xml, span);
+    write_font_weight(xml, span);
+    write_font_size(xml, span);
+
+    write_visibility(xml, span);
+    write_letter_spacing(xml, span);
+    write_word_spacing(xml, span);
+    write_text_length(xml, span);
+    write_length_adjust(xml, span);
+
+    write_smallcaps(xml, span);
+
+    write_paint_order(xml, span);
+
+    write_kerning(xml, span);
+
+    write_alignment_baseline(xml, span);
+    write_dominant_baseline(xml, span);
+
+    write_fill(&span.fill, is_clip_path, writer_context.opt, xml);
+    write_stroke(&span.stroke, writer_context.opt, xml);
+
+    let cur_text = &chunk.text[span.start..span.end];
+
+    let actual_start = *char_offset + span.start;
+    let num_chars = cur_text.chars().count();
+    let actual_end = *char_offset + span.start + num_chars;
+
+    write_coordinates(xml, text, actual_start, actual_end);
+    write_rotations(xml, text, actual_start, actual_end);
+    xml.write_text(&cur_text.replace('&', "&amp;"));
+
+    xml.end_element();
+}
+
+fn write_kerning(xml: &mut XmlWriter, span: &TextSpan) {
+    if !span.apply_kerning {
+        xml.write_attribute_raw("style", |buf| {
+            buf.extend_from_slice("font-kerning:none".as_bytes())
+        });
+    }
+}
+
+fn write_paint_order(xml: &mut XmlWriter, span: &TextSpan) {
+    if span.paint_order == PaintOrder::StrokeAndFill {
+        xml.write_svg_attribute(AId::PaintOrder, "stroke fill");
+    }
+}
+
+fn write_smallcaps(xml: &mut XmlWriter, span: &TextSpan) {
```

Comment:
I'm not sure we need such small functions, especially when they will be called only once. It only bloats the code.